flink-kubernetes/src/test/java/org/apache/flink/kubernetes/KubernetesResourceManagerTest.java

```diff
@@ -0,0 +1,372 @@
+		terminatePod(pod2);
+		resourceManager.onError(Collections.singletonList(new KubernetesPod(flinkConfig, pod2)));
+		final Pod pod3 = client.pods().list().getItems().get(0);
+		assertEquals(taskManagerPrefix + 3, pod3.getMetadata().getName());
+
+		// Delete the pod.
+		resourceManager.onAdded(Collections.singletonList(new KubernetesPod(flinkConfig, pod3)));
+		terminatePod(pod3);
+		resourceManager.onDeleted(Collections.singletonList(new KubernetesPod(flinkConfig, pod3)));
+		assertEquals(taskManagerPrefix + 4, client.pods().list().getItems().get(0).getMetadata().getName());
+	}
+
+	@Test
+	public void testGetWorkerNodesFromPreviousAttempts() throws Exception {
+		// Prepare pod of previous attempt
+		final String previewPodName = CLUSTER_ID + "-taskmanager-1-1";
+		flinkKubeClient.createTaskManagerPod(new TaskManagerPodParameter(
+			previewPodName,
+			new ArrayList<>(),
+			1024,
+			1,
+			new HashMap<>()));
+		final KubernetesClient client = getKubeClient();
+		assertEquals(1, client.pods().list().getItems().size());
+
+		// Call initialize method to recover worker nodes from previous attempt.
+		resourceManager.initialize();
+
+		final String taskManagerPrefix = CLUSTER_ID + "-taskmanager-";
+		// Register the previous taskmanager, no new pod should be created
+		registerTaskExecutor(new ResourceID(previewPodName));
+		registerSlotRequest(resourceManager);
+		assertEquals(1, client.pods().list().getItems().size());
+
+		// Register a new slot request, a new taskmanger pod will be created with attempt2
+		registerSlotRequest(resourceManager);
+		assertEquals(2, client.pods().list().getItems().size());
+		assertThat(client.pods().list().getItems().stream()
+				.map(e -> e.getMetadata().getName())
+				.collect(Collectors.toList()),
+			Matchers.containsInAnyOrder(taskManagerPrefix + "1-1", taskManagerPrefix + "2-1"));
+
+		resourceManager.getRpcService().stopService().get();
+	}
+
+	private TestingKubernetesResourceManager createAndStartResourceManager(Configuration configuration) throws Exception {
+
+		final TestingRpcService rpcService = new TestingRpcService(configuration);
+		MockResourceManagerRuntimeServices rmServices = new MockResourceManagerRuntimeServices(rpcService, TIMEOUT);
+
+		final TestingKubernetesResourceManager kubernetesResourceManager = new TestingKubernetesResourceManager(
+			rpcService,
+			"kubernetesResourceManager",
+			ResourceID.generate(),
+			configuration,
+			rmServices.highAvailabilityServices,
+			rmServices.heartbeatServices,
+			rmServices.slotManager,
+			rmServices.jobLeaderIdService,
+			new ClusterInformation("localhost", 1234),
+			testingFatalErrorHandler,
+			UnregisteredMetricGroups.createUnregisteredResourceManagerMetricGroup()
+		);
+		kubernetesResourceManager.start();
+		rmServices.grantLeadership();
+		return kubernetesResourceManager;
+	}
+
+	private void registerSlotRequest(TestingKubernetesResourceManager resourceManager) throws Exception {
```

Comment:
no need to pass this argument because `resourceManager` already class level field.

Reply:
Will fix it.